src/d.ts

```diff
@@ -48,6 +48,7 @@ export function isHNode(child: DNode): child is HNode {
  */
 export function decorate(dNodes: DNode, modifier: (dNode: DNode) => void, predicate?: (dNode: DNode) => boolean): DNode;
 export function decorate(dNodes: DNode[], modifier: (dNode: DNode) => void, predicate?: (dNode: DNode) => boolean): DNode[];
+export function decorate(dNodes: DNode | DNode[], modifier: (dNode: DNode) => void, predicate?: (dNode: DNode) => boolean): DNode | DNode[];
```

Comment:
Allow decorate to operate on a node that is typed as `DNode | DNode[]`, previous API only allowed the node to be typed as only one or the other